src/Symfony/Component/Serializer/Mapping/Factory/ClassMetadataFactory.php

```diff
@@ -44,6 +46,10 @@ public function __construct(LoaderInterface $loader, Cache $cache = null)
     {
         $this->loader = $loader;
         $this->cache = $cache;
+
+        if (null !== $cache) {
+            @trigger_error('Passing a Doctrine Cache instance as 2nd parameter of the "Symfony\Component\Serializer\Mapping\Factory\ClassMetadataFactoryInterface" is deprecated. This parameter will be removed in Symfony 4.0. Use the "Symfony\Component\Serializer\Mapping\Factory\CacheMetadataFactory" class instead.', E_USER_DEPRECATED);
```

Comment:
It seems that you are as tired as I am 😉 The second FQN must be the new one (and BTW you can use sprintf + __CLASS__ / ::class)